File: Java/src/CountFrequencies.java
import java.util.HashMap;
import java.util.Map;

public class CountFrequencies {
    public static void main(String[] args) {
        String[] words = {"apple", "banana", "apple", "orange", "banana", "apple"};
        countFrequencies(words);
    }

    public static void countFrequencies(String[] word) {
        Map<String, Integer> frequencyMap = new HashMap<>();

        for (String w : word) {
            frequencyMap.put(w, frequencyMap.getOrDefault(w, 0) + 1);
        }

        for (Map.Entry<String, Integer> entry : frequencyMap.entrySet()) {
            System.out.println(entry.getKey() + " = " + entry.getValue());
        }

    }
}

File: Java/kotlin/CountFrequencies.kt
package kotlin

object CountFrequencies {
    @JvmStatic
    fun main(args: Array<String>) {
        val words = arrayOf("apple", "banana", "apple", "orange", "banana", "apple")
        countFrequencies(words)
    }

    fun countFrequencies(word: Array<String>) {
        val frequencyMap = mutableMapOf<String, Int>()

        for (w in word) {
            frequencyMap[w] = frequencyMap.getOrDefault(w, 0) + 1
        }

        for ((key, value) in frequencyMap) {
            println("$key = $value")
        }
    }
}

